JASP-Desktop/analysis/jaspcontrolbase.h

```diff
@@ -5,34 +5,39 @@
-	Q_PROPERTY( QQuickItem*		childControlsArea	READ childControlsArea	WRITE setChildControlsArea										)
-	Q_PROPERTY( QQuickItem*		section				READ section			WRITE setSection												)
-	Q_PROPERTY( QQuickItem*		parentListView		READ parentListView									NOTIFY parentListViewChanged		)
-	Q_PROPERTY( QQuickItem*		innerControl		READ innerControl		WRITE setInnerControl		NOTIFY innerControlChanged			)
-	Q_PROPERTY( QQuickItem*		background			READ background			WRITE setBackground			NOTIFY backgroundChanged			)
-
-	Q_PROPERTY( QQmlListProperty<QQmlComponent> rowComponents READ rowComponents)
-	Q_PROPERTY( QQmlComponent * rowComponent		READ rowComponent		WRITE setRowComponent		NOTIFY rowComponentChanged			)
-	Q_PROPERTY( bool runOnChange					READ runOnChange		WRITE setRunOnChange		NOTIFY runOnChangeChanged			)
+	typedef QQmlListProperty<QQmlComponent> QQmlComponents;
+
+	Q_PROPERTY( ControlType			controlType			READ controlType		WRITE setControlType											)
+	Q_PROPERTY( QString				name				READ name				WRITE setName				NOTIFY nameChanged					)
+	Q_PROPERTY( QString				title				READ title				WRITE setTitle				NOTIFY titleChanged					) //Basically whatever a human sees on their screen when they look at this specific item.
+	Q_PROPERTY( QString				info				READ info				WRITE setInfo				NOTIFY infoChanged					)
+	Q_PROPERTY( QString				helpMD				READ helpMD											NOTIFY helpMDChanged				)
+	Q_PROPERTY( bool				isBound				READ isBound			WRITE setIsBound			NOTIFY isBoundChanged				)
+	Q_PROPERTY( bool				debug				READ debug				WRITE setDebug				NOTIFY debugChanged					)
+	Q_PROPERTY( bool				parentDebug			READ parentDebug									NOTIFY parentDebugChanged			)
+	Q_PROPERTY( bool				focusOnTab			READ focusOnTab			WRITE setFocusOnTab			NOTIFY focusOnTabChanged			)
+	Q_PROPERTY( bool				hasError			READ hasError			WRITE setHasError			NOTIFY hasErrorChanged				)
+	Q_PROPERTY( bool				hasWarning			READ hasWarning			WRITE setHasWarning			NOTIFY hasWarningChanged			)
+	Q_PROPERTY( bool				runOnChange			READ runOnChange		WRITE setRunOnChange		NOTIFY runOnChangeChanged			)
+	Q_PROPERTY( QQuickItem		*	childControlsArea	READ childControlsArea	WRITE setChildControlsArea										)
+	Q_PROPERTY( QQuickItem		*	section				READ section			WRITE setSection												)
+	Q_PROPERTY( QQuickItem		*	parentListView		READ parentListView									NOTIFY parentListViewChanged		)
+	Q_PROPERTY( QQuickItem		*	innerControl		READ innerControl		WRITE setInnerControl		NOTIFY innerControlChanged			)
+	Q_PROPERTY( QQuickItem		*	background			READ background			WRITE setBackground			NOTIFY backgroundChanged			)
+	Q_PROPERTY( QQmlComponent	*	rowComponent		READ rowComponent		WRITE setRowComponent		NOTIFY rowComponentChanged			)
+	Q_PROPERTY( QQmlComponents		rowComponents		READ rowComponents)
 
 
 public:
+	// Any addition here should also be added manually to ControlTypeToFriendlyString... I couldnt get this to work with DECLARE_ENUM...
```

Comment:
Well, I was doubting how to call the property but that is what `title` is right? 
We do this on more objects (like Analysis or jaspObjects) where we have a `name` that is non-translated and simple(ish) and a `title` that is shown to the user. So I figured this would be a good name for it here as well.

Reply:
I think you are mistaken: title is not the same as the controlTypeFriendlyName...
What I actually thought is in fact to have controlTypeFriendlyName as a virtual function of JASPControlWrapper, overriden by each sub-class, that returns a string static class member.

Reply:
Like the one I already made on JASPControlWrapper you mean?
I will change it a bit as you said because I can see it work better. But only if im assured there is always a wrapper for a control. Because a fully static one won't work because of `BoundQMLTextInput::friendlyName()`

Reply:
Better like this?

Reply:
https://github.com/jasp-stats/jasp-desktop/pull/4076/commits/5daffd09c5aa003a16ba2043982385a85a3c11ef